main.go

```diff
@@ -18,11 +18,217 @@
+
+		req, err := http.NewRequest("POST", "http://caduceus:6000/api/v3/notify", &buffer)
+		if err != nil {
+			logging.Error(logger).Log(logging.MessageKey(), "failed to create new request", logging.ErrorKey(), err.Error())
+			return err
+		}
+		req.Header.Add("Content-type", "application/msgpack")
+
+		authValue, err := acquirer.Acquire()
+		if err != nil {
+			logging.Error(logger).Log(logging.MessageKey(), "failed to acquire", logging.ErrorKey(), err.Error())
+			return err
+		}
+
+		req.Header.Add("Authorization", authValue)
+
+		resp, err := http.DefaultClient.Do(req)
+
+		if err != nil {
+			logging.Error(logger).Log(logging.MessageKey(), "failed while making HTTP request", logging.ErrorKey(), err.Error())
+			return err
+		}
+
+		defer resp.Body.Close()
+
+		// logging.Info(logger).Log(logging.MessageKey(), "finished sending event")
+
+		return err
+	}
+
+}
+
 func main() {
-	time.Sleep(5 * time.Second)
-	os.Exit(0)
+
+	var (
+		f, v                                     = pflag.NewFlagSet(applicationName, pflag.ContinueOnError), viper.New()
+		logger, metricsRegistry, caduceator, err = server.Initialize(applicationName, os.Args, f, v, basculechecks.Metrics, basculemetrics.Metrics)
+	)
+
+	// use constant secret for hash
+	secretGetter := secretGetter.NewConstantSecret("secret1234")
+
+	// set up the middleware
+	htf, err := hashTokenFactory.New("Sha1", sha1.New, secretGetter)
+	if err != nil {
+		logging.Error(logger).Log(logging.MessageKey(), "failed to setup hash token factory", logging.ErrorKey(), err.Error())
+		os.Exit(1)
+	}
+	authConstructor := basculehttp.NewConstructor(
+		basculehttp.WithTokenFactory("Sha1", htf),
+		basculehttp.WithHeaderName("X-Webpa-Signature"),
+		basculehttp.WithHeaderDelimiter("="),
+	)
+	handler := alice.New(authConstructor)
+
+	// set up the registerer
+	basicConfig := webhookClient.BasicConfig{
+		Timeout:         5 * time.Second,
+		RegistrationURL: "http://caduceus:6000/hook",
+		Request: webhook.W{
+			Config: webhook.Config{
+				URL: "http://caduceator:5000/events",
+			},
+			Events:     []string{"device-status.*"},
+			FailureURL: "http://caduceator:5000/cutoff",
+		},
+	}
+
+	acquirer, err := acquire.NewFixedAuthAcquirer("Basic dXNlcjpwYXNz")
+	if err != nil {
+		logging.Error(logger).Log(logging.MessageKey(), "failed to create basic auth plain text acquirer:", logging.ErrorKey(), err.Error())
+		os.Exit(1)
+	}
+
+	registerer, err := webhookClient.NewBasicRegisterer(acquirer, secretGetter, basicConfig)
+	if err != nil {
+		logging.Error(logger).Log(logging.MessageKey(), "failed to setup registerer", logging.ErrorKey(), err.Error())
+		os.Exit(1)
+	}
+	periodicRegisterer := webhookClient.NewPeriodicRegisterer(registerer, 4*time.Minute, logger)
+
+	// start the registerer
+	periodicRegisterer.Start()
+
+	router := mux.NewRouter()
+
+	app := &App{}
+
+	// start listening
+	router.Handle("/events", handler.ThenFunc(app.receiveEvents))
+	router.Handle("/cutoff", handler.ThenFunc(app.receiveCutoff))
```

Comment:
The handler with the authConstructor stuff shouldn't be used here.  When we receive a cutoff notification, it won't be the same as an event request.  The secret isn't involved, and the body will contain this struct marshaled as a json:
https://github.com/xmidt-org/caduceus/blob/55c30598bff0e1ac39839de130d990fe0cf28f75/outboundSender.go#L69